tests/test_modules.py

```diff
@@ -249,3 +265,62 @@ def test_module_v2_source_path_for_v1(snippetcompiler) -> None:
 
     source: module.ModuleV2Source = module.ModuleV2Source(urls=[])
     assert source.path_for("std") is None
+
+
+def test_module_v2_from_v1_path(modules_v2_dir: str, snippetcompiler) -> None:
+    """
+    Verify that attempting to load a v2 module from the v1 modules path fails with an appropriate message when a v2 module is
+    found in a v1 module source.
+    """
+    with pytest.raises(module.ModuleLoadingException) as excinfo:
+        snippetcompiler.setup_for_snippet("import minimalv2module", add_to_module_path=[modules_v2_dir])
+    cause: CompilerException = excinfo.value.__cause__
+    root_cause: Optional[BaseException] = cause.__cause__
+    assert root_cause is not None
+    assert isinstance(root_cause, module.InvalidModuleException)
+    assert root_cause.msg == (
+        "Module at %s looks like a v2 module. Please add it as a v2 requirement with `inmanta modules add <module_name>`"
```

Comment:
perhaps it would be good to also extend the testcase to include the fix proposed by the error message?

(I also wonder in this case if inmanta modules add is going to help at all?)

Reply:
It does. Why would you think it wouldn't?
I've extended both testcases though the second one is still failing because of #3321.

Reply:
because it doesn't work if the v2 module is not in a configured pip index.